yktrace/src/hwt/mapper.rs

```diff
@@ -2,73 +2,68 @@ use object::Object;
 use phdrs::objects;
 
 use crate::SirLoc;
-use hwtracer::Trace;
+use hwtracer::{HWTracerError, Trace};
+use lazy_static::lazy_static;
 use std::{borrow, env, fs};
 
-pub struct HWTMapper {
-    phdr_offset: u64,
+lazy_static! {
     /// Maps a label address to its symbol name and block index.
     ///
     /// We use a vector here since we never actually look up entries by address; we only iterate
     /// over the labels checking if each address is within the range of a block.
-    labels: Option<Vec<(u64, (String, u32))>>
+    ///
+    /// The labels are the same for each trace, and they are immutable, so it makes sense for this
+    /// to be a lazy static, loaded only once and shared.
+    ///
+    /// FIXME if we want to support dlopen(), we will have to rethink this.
+    static ref LABELS: Vec<(u64, (String, u32))> = extract_labels().unwrap();
```

Comment:
Hmm, I must admit that I'm not convinced by this. The labels don't really feel like a global variable (as the dlopen possibility suggests). Is there somewhere better we can attach them to e.g. `HWTMapper`?

Reply:
I don't think we should over-think this.

No part of Yorick supports dlopen right now, not just the mapper.

Reply:
I agree with not over-thinking, but global variables are very dangerous: if you ever want to make them non-local, you often find you're missing the plumbing necessary to pass them where needed, and that problem invariably gets worse over time. Put another way, I think we should only use `lazy_static` in exceptional cases.

Reply:
OK.

I don't want to think about this now. So I think it's best to close the PR.

Reply:
Note that SIR is loaded in a similar manner, using a global immutable lazy_static.

The SIR (like the labels) is assumed immutable, as we rely on static linkage to collect these kinds of things at compile time.

Reply:
> I don't want to think about this now. So I think it's best to close the PR.

Just to be clear: I'm not saying we shouldn't merge this PR, because, as far as I can tell, it doesn't introduce *new* global variables, right? But I think we should be more cautious in the future about introducing new global variables.

Reply:
It adds a global variable as a mechanism to share the labels between all instances of a mapper. Before each new mapper re-loaded the labels from disk.

Since we don't have a "god struct" where we can store such things, I don't know where else to store this stufff other than in a lazy static.

Reply:
Ok, so I misread things. Before this PR labels weren't stored in a global variable, but this PR turns them into a global variable. Dumb question: how did we manage before this PR?

Reply:
> Before each new mapper re-loaded the labels from disk.

Into a local variable.